[Shipping settings] Promote WooCommerce Shipping (replaces WCS&T)

kallehauge opened the pull request:
## Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

## Summary

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

This PR aims to promote WooCommerce Shipping instead of WooCommerce Shipping & Tax in the promotional banner(s) on the core shipping settings page (`/wp-admin/admin.php?page=wc-settings&tab=shipping`).

We already introduced some conditions in #48701 that would hide the banner if WC Shipping was activated but a couple of things have changed since:
1. We wish to promote `woocommerce-shipping` instead of `woocommerce-services` as the go-to shipping plugin.
2. We have achieved "parallel support" in WCS&T (read: `woocommerce-services`) so it can co-exist with WooCommerce Shipping; that way merchants can continue to use the tax part of WCS&T.

Additionally, we're also updating the shipping copy that mentions USPS and DHL Express to include UPS since WooCommerce Shipping supports UPS.

_Technical note: this PR touches on 5 different components: 1) Shipping settings guided tour, 2) "old" promotional banner, 3) experimental/new promotional banner, 4) the task list the experimental banner redirects to, and 5) the "Select your shipping options" home task._
_You can see screenshots of them all below. The reason we're touching all 5 in this PR is that they're all somewhat intertwined with each other and if only 1 of them were merged, then another one might not work as intended._

## Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

#### Primary changes:

1. Repurpose the "old" WooCommerce Shipping promotional banner on `/wp-admin/admin.php?page=wc-settings&tab=shipping` so it installs `woocommerce-shipping` instead of `woocommerce-services`.
2. Repurpose the "experimental" WooCommerce Shipping promotional banner on `/wp-admin/admin.php?page=wc-settings&tab=shipping` so it installs `woocommerce-shipping` instead of `woocommerce-services`.
3. Change the plugin we install in the hidden task list the experimental banner redirects to (`/wp-admin/admin.php?page=wc-admin&task=shipping-recommendation`)
4. Update the core shipping settings page guided tour, so it correctly listens for the `woocommerce-shipping` submenu link
5. Remove all `woocommerce-tax` limiting conditions for our shipping recommendations.
6. Remove all `woocommerce-services` limiting conditions for our shipping recommendations.

#### Secondary changes:

1. Renaming existing files and component from `services` to `shipping`. We do this since the services plugin will take over for taxes, so the naming will have to be used for a different purpose in the near future - also, it didn't reflect what happens now.
2. Add "UPS" to the different shipping label copy that mentions carriers so it correctly reflect [WooCommerce Shipping's plugin page information](https://wordpress.org/plugins/woocommerce-shipping).

### Related issues and PRs

* Resolves #51858
* Resolves #55173
* Related to #48701

<!-- Begin testing instructions -->

## How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

### Old recommended shipping plugin promotional banner

1. Install WooCommerce locally and set it up as a US store (e.g. [use A8C's address](https://automattic.com/contact))
    * _Tip: you'll have to create a WPCOM connection, so if your Woo installation isn't reachable from the internet, then you should be able to do this with a Jurassic Ninja site where you spin up this branch using the `WooCommerce Beta Tester` plugin._
2. The way to differentiate if the old or new banner should be used depends on the `shipping-smart-defaults` admin feature. Copy the snippet below to your site so we remove the flag and use the old banner _(tip: you can use the [Code Snippets plugin](https://wordpress.org/plugins/code-snippets/))_
3. Go to `/wp-admin/admin.php?page=wc-settings&tab=shipping`
4. You should now see the promotional banner at the bottom of the page (see screenshot section for example)
5. Verify the copy includes `UPS` (see screenshot section for example)
7. Click the "Get started" button
8. Verify you get a snackbar notification.
9. Click the `Finish the setup by connecting your store to WordPress.com` link in the snackbar.
10. Verify you were redirected to the WooCommerce Shipping settings page (this indirectly verifies that it installed the correct plugin as well).
11. If you're not doing the **bonus** test steps below, then remove the PHP snippet again since we're done with the "old" banner and the rest of the tests relies on the `shipping-smart-defaults` admin feature.

```php
add_filter( 'woocommerce_admin_features', function( $features ) {
	foreach( $features as $key => $feature ) {
		if ( 'shipping-smart-defaults' === $feature) {
			unset( $features[ $key ] );
		}
	}
	return $features;
}, 999, 1 );
```

#### Bonus: we should still show the banner if WooCommerce Shipping & Tax is installed and/or activate 

21. Go to the plugins page (`/wp-admin/plugins.php`)
22. Deactivate WooCommerce Shipping
23. Go to the plugins page and add a new plugin (`/wp-admin/plugin-install.php`)
24. Search for `WooCommerce Shipping & Tax`
25. Install and activate the plugin
26. Go to the core shipping settings page (`/wp-admin/admin.php?page=wc-settings&tab=shipping`)
27. Verify that we still show the promotional banner
28. Remove the PHP snippet again since we're done with the "old" banner and the rest of the tests relies on the `shipping-smart-defaults` admin feature

### Guided tour

1. Can you reuse the website from previous test, but feel free to create a fresh install as well. Just make sure the store's country is set to US (e.g. [use A8C's address](https://automattic.com/contact))
    * _Tip: you'll have to create a WPCOM connection, so if your Woo installation isn't reachable from the internet, then you should be able to do this with a Jurassic Ninja site where you spin up this branch using the `WooCommerce Beta Tester` plugin._
2. Deactivate WooCommerce Shipping if it's active.
3. The guided tour will only be shown once, so if it's not the first time you're visiting the page, then you need to set the `woocommerce_admin_reviewed_default_shipping_zones` option to `no` (wp cli: `wp option set woocommerce_admin_reviewed_default_shipping_zones no`).
4. Go to the WooCommerce home (`/wp-admin/admin.php?page=wc-admin`).
5. Click the "Review shipping options" task in the "Things to do next" section.
6. You should now see a guided tour with 3 steps. The last one pointing to the banner recommending WooCommerce Shipping.
7. Active the WooCommerce Shipping plugin
8. Set the `woocommerce_admin_reviewed_default_shipping_zones` option to `no` so we can trigger the tour again (wp cli: `wp option set woocommerce_admin_reviewed_default_shipping_zones no`).
9. You should see 3 steps again, but the last step should highlight the `WooCommerce Shipping` settings link.
10. When you reach the 3. step, verify that the copy also includes `UPS` (see the screenshot section).

### Experimental promotional banner

1. Can you reuse the website from previous test, but feel free to create a fresh install as well. Just make sure the store's country is set to US (e.g. [use A8C's address](https://automattic.com/contact))
    * _Tip: you'll have to create a WPCOM connection, so if your Woo installation isn't reachable from the internet, then you should be able to do this with a Jurassic Ninja site where you spin up this branch using the `WooCommerce Beta Tester` plugin._
2. Deactivate WooCommerce Shipping if it's active.
3. If you already have a WPCOM connection, then you need to sever it.
    * _Tip: you can install the Jetpack plugin and go to `/wp-admin/admin.php?page=my-jetpack`. At the bottom of the page you'll find a "Connection" section with an underlined link called "Manage". Press "Disconnect Jetpack" and follow the steps._
4. Go to `/wp-admin/admin.php?page=wc-settings&tab=shipping`
5. You should now see the promotional banner at the bottom of the page (see screenshot section for example)
6. Verify the copy includes `UPS` (see screenshot section for example)
7. Click the "Get started" button (it might also say "Activate" if you have the shipping plugin installed, but not activated)
8. You should now be redirected to an isolated task list (`/wp-admin/admin.php?page=wc-admin&task=shipping-recommendation`)
9. Assuming you've already define the stores location in Woo settings; press "Install & enable"
10. Press the "Connect" button
12. Verify that the WooDNA page says "WooCommerce Shipping" and not "WooCommerce Shipping & Tax"
13. Approve the connection
14. When you get back to the page, then you should be redirected to WooCommerce Shipping's settings page (`/wp-admin/admin.php?page=wc-settings&tab=shipping&section=woocommerce-shipping-settings`)
    * _The settings page will show a new connect component but that's outside the scope of this task (e.g. #51863)_
15. Go to the plugins page (`/wp-admin/plugins.php`)
16. _Optional: if you're new to WCS&T and WC Shipping, then it's a good idea to verify that you've installed WooCommerce Shipping and not WooCommerce Shipping & Tax by looking at the section parameter in the URL for the settings page. It should be `woocommerce-shipping-settings`_
17. Deactivate WooCommerce Shipping
18. Go back to the core shipping settings page (`/wp-admin/admin.php?page=wc-settings&tab=shipping`)
19. The banner should now appear again - click the active/get started button
20. You should now be redirected to the task list again (`/wp-admin/admin.php?page=wc-admin&task=shipping-recommendation`)
21. Install & enable the plugin
22. You should now be automatically redirected to the settings page because a connection already exists

#### Bonus: show banner if WooCommerce Shipping & Tax is installed and/or active 

23. Go to the plugins page (`/wp-admin/plugins.php`)
24. Deactivate WooCommerce Shipping
25. Go to the plugins page and add a new plugin (`/wp-admin/plugin-install.php`)
26. Search for `WooCommerce Shipping & Tax`
27. Install and activate the plugin
28. Go to the core shipping settings page (`/wp-admin/admin.php?page=wc-settings&tab=shipping`)
29. Verify that we still show the promotional banner

### Testing the "Select your shipping options" task

1. We have to force the task to appear. To do so, open `wp-content/plugins/woocommerce/src/Admin/Features/OnboardingTasks/Tasks/Shipping.php` and add this line at the top of the `can_view()` method:
   ```php
   public function can_view() {
		return true;
   ```
1. We also have to disable remotely fetched configuration to display the fallback recommendations: we do this by seting the `woocommerce_show_marketplace_suggestions` option to `no` (`wp option set woocommerce_show_marketplace_suggestions no`).
1. Go to `/wp-admin/admin.php?page=wc-admin&task=shipping`
1. You should see this screen:
   <img width="714" alt="Screenshot 2024-06-20 at 16 01 09" src="https://github.com/Automattic/woocommerce.com/assets/1759681/a14baf7b-cea8-427b-9aed-408b5a2f1712">
   Click "Save shipping options".
1. You should now see a recommendation for WooCommerce Shipping (you might have to wait a sec before continuing from the previous step since it's fetched async):
   <img width="708" alt="Screenshot 2024-06-20 at 16 01 46" src="https://github.com/Automattic/woocommerce.com/assets/1759681/16ed951a-a2a1-4ac1-b78a-c7dfbfc65409">
1. Verify that the copy mentions all 3 carriers as well: USPS, UPS, and DHL.
1. Now **in another tab** open `wp-admin/plugins.php` and activate WC Shipping (having WCS&T active doesn't affect this task).
1. Return to the shipping task tab, refresh it and note that WooCommerce Shipping is no longer presented.
1. _Note: we have a separate task to "fix" it so WCCOM connections will work with WC Shipping as well, so we're not attempting to change the "Connect" step as part of this task (notice: it wasn't working 100% as expected for WCS&T either)_

<!-- End testing instructions -->

## Screenshots

### "Old" and "experimental" promotional banner

![Screenshot 2025-01-30 at 01 52 14](https://github.com/user-attachments/assets/f31d75d9-e8c2-422e-85f5-42a158128aa5)

---

### Task list

![Screenshot 2025-01-29 at 20 40 41](https://github.com/user-attachments/assets/5e22352e-37f1-4c13-993d-0b98ac7f17d5)

---

### Guided Tour

![Screenshot 2025-01-29 at 20 43 58](https://github.com/user-attachments/assets/ecd2cf9e-7e6e-4c43-bf21-8202515b2b7e)

---

## Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

kallehauge committed:
Shipping settings: start prompting woocommerce-shipping

This will effectively use all the existing flows and assets and replace woocommerce-services under the hood.

kallehauge committed:
Update shipping settings tour to reflect WC Shipping changes

kallehauge committed:
Add changelog entry

kallehauge committed:
Adjust ShippingPartnerSuggestions\DefaultShippingPartners.php test

kallehauge committed:
Align experience between experimental and non-smart recommendation items

kallehauge commented on plugins/woocommerce/client/admin/client/shipping/experimental-shipping-recommendations.tsx:
```diff
@@ -43,22 +41,14 @@ const ShippingRecommendations: React.FC = () => {
 			countryCode: getCountryCode(
 				settings.general?.woocommerce_default_country
 			),
-			isJetpackConnected: _isJetpackConnected(),
```
The task list component will also prompt the merchant to connect, so there's no reason to hide the banner if we do not have a WPCOM connection.

kallehauge committed:
Remove outdated test

We no longer limit the shipping banner based on WooCommerce Tax's activation state because it's separate plugins.

kallehauge committed:
Remove another outdated tax condition test

kallehauge committed:
Connect step should use woocommerce-shipping as its from value

kallehauge committed:
Fix linting issue

waclawjacek commented:
:eyes:

waclawjacek reviewed (dismissed):
When testing the experimental banner and my site is not connected (as in the testing instructions), I am stuck at a blank post-installation screen:


https://github.com/user-attachments/assets/1d879142-3adb-44c2-9eeb-37024828e38f

But when testing the same flow with an established WPCOM connection, it works:


https://github.com/user-attachments/assets/54a8b13e-fa7f-44f4-85f0-1f2c4d094032

Can you reproduce this on your machine?

kallehauge committed:
Merge branch 'trunk' into update/51858-settings-page-promote-wcshipping

kallehauge commented:
Ah! Great find @waclawjacek! Seems like it started acting up after one of my latest commits (3092fd69529f60a253d7659324406b2ba670a29a) 🙇 

I'll look into it and request a new review when I've found the culprit 😄

kallehauge committed:
Merge branch 'trunk' into update/51858-settings-page-promote-wcshipping

kallehauge committed:
Merge branch 'trunk' into update/51858-settings-page-promote-wcshipping

kallehauge committed:
Remove outdated plugin conditions

kallehauge committed:
WC Shipping no longer requires Jetpack

WC Shipping is able to create its own WPCOM connection, so there is no hard dependency on the plugin anymore.

kallehauge committed:
Merge branch 'trunk' into update/51858-settings-page-promote-wcshipping

kallehauge committed:
Merge branch 'trunk' into update/51858-settings-page-promote-wcshipping

kallehauge committed:
Show promotional banner if WCS&T is installed

kallehauge committed:
Remove WooCommerce Tax tests from shipping banners

kallehauge committed:
Hide default WooCommerce Shipping suggestion if it is already installed

kallehauge committed:
Dismisslist components has been changed in trunk

kallehauge committed:
Fix broken connection step for experimental banner

kallehauge committed:
Remove outdated test

Ferdev commented:
I'm taking care of this PR, considering Mr. Jacek is AFK :+1:

Ferdev reviewed (approved):
Ok, I went through all the test process and everything worked as expected, great job! (and great job doing the tutorial!) :+1:
I found one issue, not sure if related to this PR, probably not. In the WooCommerce Setup Wizard, when the different plugin recommendations are presented, the one for WooCommerce Shipping is not mentioning UPS:

![shot-2025-02-13-15-11-49](https://github.com/user-attachments/assets/bb8abd52-9fd9-45ae-8647-5a455ced1d43)

I don't think is related to this PR, but worth it to mention anyway.

The code looks fine, so I think this is good to :shipit:

kallehauge commented:
Awesome! Thank you for going through it all @Ferdev 👏 

>I found one issue, not sure if related to this PR, probably not. In the WooCommerce Setup Wizard, when the different plugin recommendations are presented, the one for WooCommerce Shipping is not mentioning UPS:
>I don't think is related to this PR, but worth it to mention anyway.

Thank you for mentioning it; better safe than sorry! It's being addressed in this PR: https://github.com/woocommerce/woocommerce/pull/55147 😄

kallehauge committed:
Merge branch 'trunk' into update/51858-settings-page-promote-wcshipping

beaulebens commented:
Can we get the multiple old logos/colors updated in here to match the new brand guidelines as well please? (separate PR)
